pillarbox-cast-receiver/src/main/java/ch/srgssr/pillarbox/cast/receiver/PillarboxMediaCommandCallback.kt

```diff
@@ -0,0 +1,339 @@
+        Log.d(TAG, "onQueueInsert $senderId ${requestData.items.size} before ${requestData.insertBefore}")
+        Log.d(TAG, "Items: ${requestData.items.map { "${it.media?.metadata?.getString(MediaMetadata.KEY_TITLE)}" }}")
+        mediaQueueManager.queueItems?.let { queueItems ->
+            val insertIndex = requestData.insertBefore?.let {
+                mediaQueueManager.getIndexOfItemIdOrNull(it)
+            } ?: Int.MAX_VALUE
+
+            insert(requestData.items, insertIndex)
+
+            val mediaItems = requestData.items.map(mediaItemConverter::toMediaItem)
+            player.addMediaItems(insertIndex, mediaItems)
+        }
+        mediaQueueManager.notifyItemsInserted(requestData.items.map { item -> item.itemId }, requestData.insertBefore)
+        mediaManager.broadcastMediaStatus()
+        return Tasks.forResult<Void?>(null)
+    }
+
+    /**
+     * https://developers.google.com/android/reference/com/google/android/gms/cast/tv/media/QueueReorderRequestData
+     */
+    override fun onQueueReorder(senderId: String?, requestData: QueueReorderRequestData): Task<Void?> {
+        Log.d(TAG, "onQueueReorder ${requestData.currentItemId} ${requestData.itemIds} before ${requestData.insertBefore}")
+        val task = Tasks.forResult<Void?>(null)
+        val queueItems = mediaQueueManager.queueItems
+        if (queueItems.isNullOrEmpty() || requestData.itemIds.isEmpty()) {
+            return task
+        }
+        reorderItems(queueItems = queueItems, requestData.itemIds, requestData.insertBefore)
+        mediaManager.broadcastMediaStatus()
+        return task
+    }
+
+    override fun onQueueRemove(senderId: String?, requestData: QueueRemoveRequestData): Task<Void?> {
+        Log.d(TAG, "onQueueRemove ${requestData.itemIds}")
+        mediaQueueManager.queueItems?.let { queueItems ->
+            val removeMediaId = mutableListOf<Int>()
+            requestData.itemIds.forEach { itemId ->
+                mediaQueueManager.getIndexOfItemIdOrNull(itemId)?.let {
+                    queueItems.removeAt(it)
+                    player.removeMediaItem(it)
+                    removeMediaId.add(itemId)
+                }
+            }
+            if (removeMediaId.isNotEmpty()) {
+                Log.d(TAG, "onRemovedItems $removeMediaId")
+                mediaQueueManager.notifyItemsRemoved(removeMediaId)
+            }
+        }
+        mediaManager.broadcastMediaStatus()
+        return Tasks.forResult<Void?>(null)
+    }
+
+    @Suppress("MaximumLineLength")
+    override fun onQueueUpdate(senderId: String?, requestData: QueueUpdateRequestData): Task<Void?> {
+        Log.d(
+            TAG,
+            "onQueueUpdate items = ${requestData.items} ${requestData.currentItemId} -> ${requestData.jump} " +
+                "${requestData.shuffle} ${requestData.repeatMode}"
+        )
+        requestData.shuffle?.let {
+            if (mediaQueueManager.queueItems.isNullOrEmpty()) return Tasks.forResult<Void?>(null)
```

Comment:
Is it fine to `return` here, without processing the other data first?

Reply:
It is fine because when suffle is set other attributs are null, it is a well designed api by google. I think it is better to return from let.